src/Recurr/Transformer/InstanceGenerator.php

```diff
@@ -0,0 +1,771 @@
+                // Iterate through all exdates until we reach the `$min_r_date`
+                while ($generator->valid() && $generator->current()->getTimestamp() < $minRDate->getTimestamp()) {
+                    $generator->next();
+                }
+
+                // If the current EXDATE matches the current `$min_r_date`, then the date should be excluded from the
+                // series.
+                if ($generator->valid() && $generator->current()->getTimestamp() === $minRDate->getTimestamp()) {
+                    $dateIsExcluded = true;
+                }
+
+                $exGeneratorsToKeep[] = $generator;
+            }
+
+            $exGenerators = $exGeneratorsToKeep;
+
+            if (!$dateIsExcluded) {
+                $iterations++;
+                yield $minRDate;
+            }
+
+            foreach ($rGenerators as $generator) {
+                // Advance all the `$r_generators` that yielded the previous result.
+                if ($generator->current()->getTimestamp() === $minRDate->getTimestamp()) {
+                    $generator->next();
+                }
+            }
+            // Keep looping so long as the `$r_generators` are yielding dates and we're under the iteration limit.
+        } while (!empty($rGenerators) && (null === $iterationLimit || $iterations < $iterationLimit));
+
+    }
+
+    /**
+     * Returns a generator that provides a given array of dates, copied with the given time zone.
+     *
+     * @param DateTime[] $datetimes The dates to be yielded by the generator.
+     * @param DateTimeZone $timeZone The timezone context to give all the yielded dates.
+     * @return Generator A generator that yields {@link DateTime} values representing instance start time.
+     */
+    private function generateFromDates(array $datetimes, DateTimeZone $timeZone)
+    {
+        $compare = function (DateTime $datetime1, DateTime $datetime2) {
+            return ($datetime1->getTimestamp() - $datetime2->getTimestamp());
+        };
+
+        usort($datetimes, $compare);
+
+        foreach ($datetimes as $date) {
+            $result = clone $date;
+            $result->setTimezone($timeZone);
+            $result->getTimestamp(); // Fixes PHP timezone bug.
+
+            yield $result;
+        }
+    }
+
+    /**
+     * Returns a generator that provides instance dates based on the given time zone.
+     *
+     * @param Rule $rule The iCal RRULE or EXRULE value.
+     * @param DateTimeZone $timeZone The timezone context to give all the yielded dates.
+     * @param bool $ignoreCounts Ignores the `COUNT` constraint of all rules given. This is useful if the `$seed`
+     *   represents a date in the middle of the series and using `COUNT` would be cause for an inaccurate end date.
+     * @return Generator A generator that yields {@link DateTime} values representing instance start time.
+     */
+    private function generateFromRule(Rule $rule, DateTimeZone $timeZone, $ignoreCounts = false)
```

Comment:
This method is just code moved from https://github.com/robinpowered/recurr/pull/1/files#diff-6ad4b944c60048c436a680c566f45994L108

The only differences are we remove the last day of month fix (non ical compliant) and yield dates instead of add them to an array that is returned at the end. The constraint logic is also removed in favor of having the call manage that during iteration/generator.